cmd/srcd/cmd/sql_test.go

```diff
@@ -0,0 +1,215 @@
+// +build integration
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	cmdtest "github.com/src-d/engine/cmd/test-utils"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type SQLTestSuite struct {
+	cmdtest.IntegrationSuite
+	testDir string
+}
+
+func TestSQLTestSuite(t *testing.T) {
+	s := SQLTestSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *SQLTestSuite) SetupTest() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "sql-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (s *SQLTestSuite) TearDownTest() {
+	s.RunStop(context.Background())
+	os.RemoveAll(s.testDir)
+}
+
+const showTablesOutput = `+--------------+
+|    TABLE     |
++--------------+
+| blobs        |
+| commit_blobs |
+| commit_files |
+| commit_trees |
+| commits      |
+| files        |
+| ref_commits  |
+| refs         |
+| remotes      |
+| repositories |
+| tree_entries |
++--------------+
+`
+
+func (s *SQLTestSuite) TestInit() {
+	require := s.Require()
+
+	repoPath := filepath.Join(s.testDir, "reponame")
+	err := os.Mkdir(repoPath, os.ModePerm)
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+
+	_, err = s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	buf, err := s.RunSQL(context.TODO(), "select * from repositories")
+	require.NoError(err)
+
+	expected := `+---------------+
+| REPOSITORY ID |
++---------------+
+| reponame      |
++---------------+
+`
+	require.Contains(buf.String(), expected)
+}
+
+func (s *SQLTestSuite) TestNoInit() {
+	require := s.Require()
+
+	buf, err := s.RunSQL(context.TODO(), "show tables")
+	require.NoError(err)
+
+	require.Contains(buf.String(), showTablesOutput)
+}
+
+func (s *SQLTestSuite) TestValidQueries() {
+	testCases := []string{
+		"show tables",
```

Comment:
isn't this guaranteed by gitbase itself? If so, why replicating tests here?

Reply:
gitbase should work for `show tables`, but we need to test that `srcd` is in fact sending the query as expected.
I put some variations to check that we don't mess with the query string. This could happen if for example tomorrow we decide to trim spaces and add a `;` to all queries, this test would detect some corner cases.

Reply:
should we rewrite the query sent? should not `gitbase` to receive it as it is? why using source{d} Engine to polish sql queries?
I agree that we need to test that `srcd` sends the query, and gets the proper result, but imo it is done just sending a query, and comparing with the expected result; caring about casing, spaces or other formatting things, should be done by `gitbase`.

Anyhow, it's not a blocker to merge this PR, just a personal opinion ;)